client/packages/common/src/ui/components/domain/Dashboard/StatsPanel.tsx

```diff
@@ -58,51 +56,89 @@ export const StatsPanel: FC<StatsPanelProps> = ({
       </Grid>
     </Grid>
   );
+};
 
-  return (
-    <Paper
-      sx={{
-        borderRadius: '16px',
-        marginTop: '14px',
-        marginBottom: '21px',
-        boxShadow: theme => theme.shadows[1],
-        padding: '14px 24px',
-        width: width ? `${width}px` : undefined,
-      }}
-    >
-      <Grid container>
-        <Grid alignItems="center" display="flex">
-          <Grid item style={{ marginInlineEnd: 8 }}>
-            <StockIcon
-              color="secondary"
-              style={{
-                height: 16,
-                width: 16,
-                fill: '#3568d4',
-              }}
-            />
-          </Grid>
-          <Grid item>
-            <Typography
-              color="secondary"
-              style={{ fontSize: 12, fontWeight: 500 }}
-            >
-              {title}
-            </Typography>
-          </Grid>
+const Content = ({
+  error,
+  isError,
+  isLoading,
+  stats,
+}: {
+  error: any;
+  isError: boolean;
+  isLoading: boolean;
+  stats: Stat[];
+}) => {
+  const t = useTranslation();
+  const isPermissionDenied = isPermissionDeniedException(error);
+
+  switch (true) {
+    case isError:
```

Comment:
this is interesting way to do early return (through switch statement vs traditional if () return), one extra level of indentation but still very readable

Reply:
yeah, bit of a novelty - I picked up the `switch (true)` from @CarlosNZ and thought it was neat, so I've used it a bit